src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -29,18 +29,20 @@ pub fn generate_lockfile(manifest_path: &Path, config: &Config)
 
 pub fn update_lockfile(manifest_path: &Path,
                        opts: &UpdateOptions) -> CargoResult<()> {
-    let package = try!(Package::for_path(manifest_path, opts.config));
-
-    let previous_resolve = match try!(ops::load_pkg_lockfile(&package,
-                                                             opts.config)) {
-        Some(resolve) => resolve,
-        None => bail!("a Cargo.lock must exist before it is updated")
-    };
 
     if opts.aggressive && opts.precise.is_some() {
         bail!("cannot specify both aggressive and precise simultaneously")
     }
 
+    let package = try!(Package::for_path(manifest_path, opts.config));
+    let path = Path::new(manifest_path);
+    if !path.parent().unwrap().join("Cargo.lock").exists(){
```

Comment:
Also, let's fold this into the `None` case of `load_pkg_lockfile` so we don't have to emit an extra syscall and we can avoid an `unwrap()` below